tests/chocolatey-tests/chocolateyProfile.Tests.ps1

```diff
@@ -19,22 +19,33 @@ Describe "Chocolatey Profile" -Tag Chocolatey, Profile, Environment {
             [bool]((Get-Command TabExpansion).ScriptBlock -match "ChocolateyTabExpansion") | Should -BeTrue
         }
 
-        It "Should list completions for Top Level Commands" {
+        It "Should list completions for all Top Level Commands, sorted alphabetically, but not aliases or unpackself" {
             $Command = "choco "
             $Completions = (TabExpansion2 -inputScript $Command -cursorColumn $Command.Length).CompletionMatches.CompletionText
 
-            # These should be first
-            $Completions[0] | Should -Be "-?"
-            $Completions[1] | Should -Be "search"
-
-            # TODO: Determine why these lines are commented out. Remove if not needed, get working otherwise.
-            <#  Completions don't contain aliases, etc
-            foreach ($Command in $(
-                $(choco --help) -match " \* (?<Command>\w+) -" -replace " \* (?<Command>\w+) -.+", '$1'
-            )) {
-                $Completions | Should -Contain $Command
-            }
-            #>
+            $sortedCompletions = $Completions | Sort-Object -Property { $_ -replace '[^a-z]' }
+            $differences = Compare-Object -ReferenceObject $sortedCompletions -DifferenceObject $Completions
+            $differences | Should -BeNullOrEmpty -Because ($differences | Format-Table | Out-String)
+
+            # These are not provided by tab completion as they are either command aliases or not intended to be used
+            # by end-users.
+            $missingFromTabCompletion = @(
+                'features'
+                'find'
+                'setapikey'
+                'sources'
+                'synchronize'
+                'templates'
+                'unpackself'
+            )
+
+            # Fail the test if any choco command listed by `choco --help` isn't either in the missing list or the
+            # tab completion.
+            $unaccountedForCommands = @(choco --help) -match " \* \w+ -" -replace " \* (?<Command>\w+) -.+", '${Command}' |
+                Where-Object { $_ -notin $missingFromTabCompletion } |
+                Where-Object { $_ -notin $Completions }

```

Comment:
~~You should come to my talk at PowerShell Summit where I explain that this doesn't work on PowerShell v2...~~

Ignore that, I got half through writing it when I realized we're in the pester tests here and this is fine 👍

Reply:
I still can't remember if it's the `-contains` or `-in` ones that aren't in PSv2. 😂

Reply:
Oh I only know it's `-in` because I was just going over my summit talk last night and it's about using git bisect to track down that commit where I learned that 😂